packages/puppeteer-core/src/api/JSHandle.ts

```diff
@@ -43,9 +49,12 @@ import {ElementHandle} from './ElementHandle.js';
  *
  * @public
  */
+@moveable
 export abstract class JSHandle<T = unknown>
-  implements Disposable, AsyncDisposable
+  implements Disposable, AsyncDisposable, Moveable
 {
+  declare move: () => this;
```

Comment:
that does not require decorators and makes polymorphism work out of the box it seems

Reply:
also I probably miss the moved=true to true in dispose handles.

Reply:
(would be good to add a test for multiple moves)

Reply:
Hmm, I just realized that this works because the `dispose` is applied to the this context. Yea, I think we can remove the element handle implementation. I'll keep the test though.